ddtrace/settings/_inferred_base_service.py

```diff
@@ -0,0 +1,203 @@
+import fnmatch
+import os
+import pathlib
+import re
+import sys
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+
+INIT_PY = "__init__.py"
+ALL_PY_FILES = "*.py"
+
+CACHE: Dict[str, Optional[str]] = {}
+
+
+class DetectionContext:
```

Comment:
do we need this indirection? do you see more context being added later?

e.g. why have a wrapper class when you can just pass `os.environ` around?

Reply:
This could just be a NamedTuple

Reply:
Done.